### PR TITLE
Removed warnings from msal-node.api.md

### PR DESCRIPTION
**Pull request by Robbie-Microsoft**
JSDocs improvements to eliminate the warnings in msal-node.api.md

### CODE REVIEW

File: lib/msal-node/apiReview/msal-node.api.md [hunk 1]
```diff
@@ -737,32 +667,6 @@ export const version = "2.15.0";
...
-// src/client/OnBehalfOfClient.ts:310:8 - (tsdoc-param-tag-missing-hyphen) The @param block should be followed by a parameter name and then a hyphen
-// src/client/PublicClientApplication.ts:310:8 - (tsdoc-param-tag-missing-hyphen) The @param block should be followed by a parameter name and then a hyphen
-// src/client/UsernamePasswordClient.ts:74:8 - (tsdoc-param-tag-missing-hyphen) The @param block should be followed by a parameter name and then a hyphen
-// src/client/UsernamePasswordClient.ts:75:8 - (tsdoc-param-tag-missing-hyphen) The @param block should be followed by a parameter name and then a hyphen
-// src/client/UsernamePasswordClient.ts:114:8 - (tsdoc-param-tag-missing-hyphen) The @param block should be followed by a parameter name and then a hyphen
 // src/index.ts:8:12 - (tsdoc-characters-after-block-tag) The token "@azure" looks like a TSDoc tag but contains an invalid character "/"; if it is not a tag, use a backslash to escape the "@"
 // src/index.ts:8:4 - (tsdoc-undefined-tag) The TSDoc tag "@module" is not defined in this configuration
```
Comment (Robbie-Microsoft) [inline lib/msal-node/apiReview/msal-node.api.md:669]: Any thoughts about how I can resolve these warnings?

File: lib/msal-node/src/packageMetadata.ts [hunk 2]
```diff
@@ -1,3 +1,4 @@
 /* eslint-disable header/header */
 export const name = "@azure/msal-node";
+/** @public */
 export const version = "2.15.0";
```
Comment (Robbie-Microsoft) [inline lib/msal-node/src/packageMetadata.ts:4]: Am I allowed to edit this file? Is this auto-generated?

**Review by bgavrilMS — APPROVED** [review]

**Review by gladjohn — APPROVED** [review]

File: lib/msal-node/src/cache/serializer/Serializer.ts [hunk 3]
```diff
@@ -20,18 +20,22 @@ import {
...
 
+/**
+ * This class serializes cache entities to be saved into in-memory object types defined internally
+ * @internal
+ */
 export class Serializer {
     /**
      * serialize the JSON blob
-     * @param data
+     * @param data - JSON blob cache
```
Comment (sameerag) [inline lib/msal-node/src/cache/serializer/Serializer.ts:30]: Nit: The description above and adding in the same line with a separator, are they making a difference in typedocs how they show up?

File: lib/msal-node/src/client/ClientApplication.ts [hunk 4]
```diff
@@ -39,7 +39,7 @@ import {
 } from "@azure/msal-common/node";
 import {
     Configuration,
-    buildAppConfiguration,
+    _buildAppConfiguration,
```
Comment (sameerag) [inline lib/msal-node/src/client/ClientApplication.ts]: Why `_`? We are limiting `_` usage to JS generated code.
Refer: [naming style](https://google.github.io/styleguide/tsguide.html#naming-style) and [generated code](https://google.github.io/styleguide/tsguide.html#generated-code)

**Review by sameerag — APPROVED** [review]

File: lib/msal-node/src/cache/serializer/Serializer.ts (hunk 3 above)
Comment (Robbie-Microsoft) [reply lib/msal-node/src/cache/serializer/Serializer.ts:30]: They are. A warning is generated when the delimiter isn't used.

File: lib/msal-node/src/packageMetadata.ts (hunk 2 above)
Comment (sameerag) [reply lib/msal-node/src/packageMetadata.ts:4]: Yes, ideally supposed to be updated every release and I see this is up-to-date in browser. We will need to do the same for node.

File: lib/msal-node/apiReview/msal-node.api.md (hunk 1 above)
Comment (sameerag) [reply lib/msal-node/apiReview/msal-node.api.md:669]: Ignore them for now, will need to check if Hector can add a rule that will void these warnings. cc @hectormmg

**Pull request merged**